pkgs/applications/misc/navit/default.nix

```diff
@@ -21,34 +31,54 @@ stdenv.mkDerivation rec {
     sha256 = "0vg6b6rhsa2cxqj4rbhfhhfss71syhnfa6f1jg2i2d7l88dm5x7d";
   };
 
-  #hardeningDisable = [ "format" ];
-  NIX_CFLAGS_COMPILE = [ "-I${SDL.dev}/include/SDL" ];
+  patches = [ ./CMakeLists.txt.patch ];
+  hardeningDisable = [ "format" ];
+
+  NIX_CFLAGS_COMPILE = [ ]
+    ++ optional sdlSupport "-I${SDL.dev}/include/SDL"
+    ++ optional speechdSupport "-I${speechd}/include/speech-dispatcher";
 
-  # TODO: fix speech options.
-  cmakeFlags = [ "-DSAMPLE_MAP=n " "-DCMAKE_BUILD_TYPE=RelWithDebInfo" "-Dsupport/espeak=FALSE" "-Dspeech/qt5_espeak=FALSE" ];
+  # we choose only cmdline and speech-dispatcher speech options.
+  # espeak builtins is made for non-cmdline OS as winCE
+  cmakeFlags = [
+    "-DSAMPLE_MAP=n " "-DCMAKE_BUILD_TYPE=RelWithDebInfo"
+    "-Dspeech/qt5_espeak=FALSE" "-Dsupport/espeak=FALSE"
+  ];
 
-  buildInputs = [ gtk2 SDL fontconfig freetype imlib2 SDL_image libGLU_combined freeimage libxslt
-    libXmu freeglut python gettext quesoglc gd postgresql qtbase SDL_ttf fribidi pcre qtquickcontrols
-    espeak qtmultimedia qtspeech qtsensors qtlocation qtdeclarative qtsvg dbus-glib librsvg shapelib glib 
-    cairo gdk_pixbuf pango atk ];
+  buildInputs = [
+    gtk2 fontconfig freetype imlib2 libGLU_combined freeimage
+    libxslt libXmu freeglut python gettext quesoglc gd
+    fribidi pcre  dbus dbus-glib librsvg shapelib glib
+    cairo gdk_pixbuf pango atk
+  ] ++ optionals sdlSupport [ SDL SDL_ttf SDL_image ]
+    ++ optional postgresqlSupport postgresql
+    ++ optional speechdSupport speechd
+    ++ optionals qtSupport [
+      qtquickcontrols qtmultimedia qtspeech qtsensors
+      qtbase qtlocation qtdeclarative qtsvg
+  ];
 
-  nativeBuildInputs = [ pkgconfig cmake patchelf bzip2 ];
+  nativeBuildInputs = [ makeWrapper pkgconfig cmake patchelf bzip2 ];
 
   # we dont want blank screen by defaut
   postInstall = ''
+    # copy a better configuration for nixos desktop
+    cp "${./navit.xml}" $out/share/navit/
     # emulate DSAMPLE_MAP
-    mkdir -p $out/share/navit/maps/maps
+    mkdir -p $out/share/navit/maps/
     bzcat "${sample_map}" | $out/bin/maptool "$out/share/navit/maps/osm_bbox_11.3,47.9,11.7,48.2.bin"
   '';
 
   # TODO: fix upstream?
   postFixup = ''
     for lib in $(find "$out/lib/navit/" -iname "*.so" ); do
-      patchelf --set-rpath ${stdenv.lib.makeLibraryPath buildInputs} $lib
+      patchelf --set-rpath ${makeLibraryPath buildInputs} $lib
     done
+    wrapProgram $out/bin/navit \
+      --prefix PATH : ${makeBinPath ([ espeak xkbd ] ++ optional speechdSupport [ speechd ]) }
```

Comment:
Why is `[ espeak xkbd ]` necessary now? Should that be optional?

Reply:
i don't make that optional since i think a gps navitation has to have it. It's forbidden to look at screen when we drive here. i can make it optional, but previously, i didn't.

Reply:
Well if it is an optional dependency according to upstream, I think it is better practice to make it optional. If you think it is essential, you can then default the optional dependency to `true`.